InternTrack.Core.Api/Services/Foundations/Interns/InternService.cs

```diff
@@ -36,6 +36,20 @@ public partial class InternService : IInternService
             return await this.storageBroker.InsertInternAsync(intern);
         });
 
+        public ValueTask<Intern> ModifyInternAsync(Intern intern) =>
+        TryCatch(async () =>
+        {
+            ValidateInternOnModify(intern);
+
+            Intern maybeIntern =
+            await this.storageBroker.SelectInternByIdAsync(intern.Id);
+
+            ValidateStorageIntern(maybeIntern, intern.Id);
```

Comment:
I suggest to rename `ValidateStorageIntern` to -> `ValidateStorageInternExists` and call it inside `ValidateAgainstStorageInternOnModify`. It would be consisted as `ValidateInternOnModify` call previous to this.